README.md

```diff
@@ -72,11 +80,88 @@ building with 4 cores can be done with the following command:
 
   `make -j4`
 
-### 4. Running Models
+### 4. Using Dataloaders
+
+This repository provides dataloaders and data preprocessing modules for mlpack library.
+It also provides utility function required required for downloading, extracting and processing
+image, text and sequential data. For more information about dataloaders and utility functions,
+Refer to our wiki page.
+
+#### 1. Dataloaders for popular datasets.
+
+Creating and processing data can be done in just a single line. Don't have the dataset downloaded,
+No worries, we will download and preprocess it for you. Kindly refer to sample code given below.
+
+```cpp
+const string datasetName = "mnist";
+bool shuffleData = true;
+double ratioForTrainTestSplit = 0.75;
+
+// Create the DataLoader object.
+DataLoader<> dataloader(datasetName, shuffleData, ratioForTrainTestSplit);
+```
+
+To train or test your model with our dataloaders is very simple.
+```cpp
+// Use the dataloader for training.
+model.Train(dataloader.TrainFeatures(), dataloader.TrainLabels());
+ 
+// Use the dataloader for prediction.
+model.Predict(dataloader.TestFeatures(), dataloader.TestLabels());
+```
+
+Currently supported datasets are mentioned below :
+((I think we should add a table here with various columns and details.))
```

Comment:
+1
Do you think we should do that now, or in another PR?

Reply:
I can do that now. I'll push a commit in couple of minutes.

Reply:
I think the columns would be name, type, details. Does this make sense?

Reply:
Done, You could take a look at the table [here](https://github.com/kartikdutt18/models/blob/Dataloader-documentation/README.md).